coral/src/app/features/connectors/request/ConnectorEditRequest.tsx

```diff
@@ -0,0 +1,299 @@
+import {
+  Alert,
+  BorderBox,
+  Box,
+  Button,
+  Grid,
+  Label,
+  Option,
+  Typography,
+  useToast,
+} from "@aivenio/aquarium";
+import MonacoEditor from "@monaco-editor/react";
+import { useMutation, useQuery } from "@tanstack/react-query";
+import isEqual from "lodash/isEqual";
+import { useEffect, useState } from "react";
+import { SubmitHandler, Controller as _Controller } from "react-hook-form";
+import { useNavigate, useParams, useSearchParams } from "react-router-dom";
+import { Dialog } from "src/app/components/Dialog";
+import {
+  Form,
+  NativeSelect,
+  SubmitButton,
+  TextInput,
+  Textarea,
+  useForm,
+} from "src/app/components/Form";
+import {
+  ConnectorRequestFormSchema,
+  connectorRequestFormSchema,
+} from "src/app/features/connectors/request/schemas/connector-request-form";
+import { Routes } from "src/app/router_utils";
+import {
+  ConnectorDetailsForEnv,
+  editConnector,
+  getConnectorDetailsPerEnv,
+} from "src/domain/connector";
+import { HTTPError } from "src/services/api";
+import { parseErrorMsg } from "src/services/mutation-utils";
+
+function ConnectorEditRequest() {
+  const { connectorName } = useParams();
+  const [searchParams] = useSearchParams();
+  const navigate = useNavigate();
+  const toast = useToast();
+
+  const env = searchParams.get("env");
+
+  const [cancelDialogVisible, setCancelDialogVisible] = useState(false);
+
+  const form = useForm<ConnectorRequestFormSchema>({
+    schema: connectorRequestFormSchema,
+    defaultValues: {
+      connectorName,
+    },
+  });
+
+  const {
+    data: connectorDetailsForEnv,
+    isFetched: connectorDetailsForEnvIsFetched,
+  } = useQuery<ConnectorDetailsForEnv, HTTPError>(
+    ["getConnectorDetailsPerEnv", connectorName, env],
+    {
+      queryFn: () =>
+        getConnectorDetailsPerEnv({
+          connectorName: connectorName || "",
+          envSelected: env || "",
+        }),
+    }
+  );
+
+  // Handle errors when environment or topic does not exist
+  useEffect(() => {
+    if (env === null) {
+      navigate(Routes.CONNECTORS, { replace: true });
+      toast({
+        message: `No environment provided.`,
+        position: "bottom-left",
+        variant: "danger",
+      });
+      return;
+    }
+    if (
```

Comment:
Yeah it's a little tricky, because the front-end implementation is technically correct if we trust the API, but factually incorrect as you described. I think the error message is better if we add mention of the environment, though:

<img width="600" alt="Screenshot 2023-07-28 at 09 06 42" src="https://github.com/aiven/klaw/assets/20607294/c0a8db3c-e636-469d-ac0c-327e4d4ca777">

 I also noticed I didn't handle the actual errors from the query :o 
 
 [95f2899](https://github.com/aiven/klaw/pull/1523/commits/95f28990c962f5de9e092899021c53249785015e)

Reply:
Awesome! Maybe we want to make a note of this for somewhat later to improve the API here.